src/Analyzers/Core/Analyzers/AddRequiredParentheses/AbstractAddRequiredParenthesesDiagnosticAnalyzer.cs

```diff
@@ -100,14 +102,11 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
         var childPrecedenceKind = _precedenceService.GetPrecedenceKind(binaryLike);
         var parentPrecedenceKind = _precedenceService.GetPrecedenceKind(parentBinaryLike);
 
-        var childEquivalenceKey = GetEquivalenceKey(childPrecedenceKind);
-        var parentEquivalenceKey = GetEquivalenceKey(parentPrecedenceKind);
+        var collapsedChildPrecedenceKind = CollapsePrecedenceGroups(childPrecedenceKind);
+        var collapsedParentPrecedenceKind = CollapsePrecedenceGroups(parentPrecedenceKind);
 
-        // only add parentheses within the same precedence band.
-        if (childEquivalenceKey != parentEquivalenceKey)
-        {
+        if (IsClearPrecedenceBoundary())
```

Comment:
@akhera99 here.  we view the precedence as unclear.  the *fixer* then actually wraps with parens (that fixer didn't need to change, it just trusts what the analyzer reports).